### ISSUE TITLE
Interceptor for 302 responses

### ISSUE THREAD

**Issue by paul321** [mzabriskie/axios#980]
Hi,
I have a problem with 302 responses, we need to access headers and cookies after the user authenticates the response comes with set-cookie and with a 302 status for redirection, we are using nodejs with custom cookie jar and we save cookies in interceptors but there is no interceptor for 302 responses.

Is there any solution to this?

**Comment by pimlie** [comment]
Set `maxRedirects: 0` in Request Config?

**Comment by paul321** [comment]
I did that but i consider it a hack because after i get the redirect i need to manually request the location in the header

**Comment by pimlie** [comment]
Unfortunately it seems there is no other way, actually axios handles no redirects at all. The `maxRedirects` option is passed to the http(s) transport from nodejs. The http client from nodejs does the automatic redirect.

**Comment by rubennorte** [comment]
You can use `maxRedirects: 0` and a response interceptor to handle additional requests.

**Comment by matepaiva** [comment]
I arrived here because I had a similar problem: the server was redirecting the when the user was not logged in and I was trying to find a way to intercept it. But it turns out that, in my case, when I added a common header to every request telling that it is a XHR, the server started responding correctly. Just writing it down to maybe help another person with the same issue.

```
axios.defaults.headers.common['X-Requested-With'] = 'XMLHttpRequest';
```
